tests/test_internals/test_movie_utils.py

```diff
@@ -110,3 +115,99 @@ def test_isopened_assertions(self):
         self.assertRaises(AssertionError, vcc.get_movie_frame_count)
         self.assertRaises(AssertionError, vcc.get_movie_frame_dtype)
         self.assertRaises(AssertionError, vcc.__next__)
+
+
+@unittest.skipIf(not CV2_INSTALLED, "cv2 not installed")
+class TestMovieInterface(unittest.TestCase):
+
+    frame_shape = (800, 600, 3)
+    number_of_frames = 500
+    fps = 25
+
+    def setUp(self):
+        self.test_dir = tempfile.mkdtemp()
+        self.nwbfile = make_nwbfile_from_metadata(dict(NWBFile=dict(session_start_time=datetime.now())))
+        self.nwbfile_path = os.path.join(self.test_dir, "movie_test.nwb")
+
+    def create_movie(self, fps, frame_shape, number_of_frames):
+        movie_frames = np.random.randint(0, 255, size=[number_of_frames, *frame_shape], dtype="uint8")
+        movie_file = os.path.join(self.test_dir, "test.avi")
+        writer = cv2.VideoWriter(
+            filename=movie_file,
+            apiPreference=None,
+            fourcc=cv2.VideoWriter_fourcc(*"HFYU"),
+            fps=fps,
+            frameSize=frame_shape[1::-1],
+            params=None,
+        )
+        for k in range(number_of_frames):
+            writer.write(movie_frames[k, :, :, :])
+        writer.release()
+        return movie_file
+
+    def test_iterator_general(self):
+        movie_file = self.create_movie(self.fps, self.frame_shape, self.number_of_frames)
+        it = H5DataIO(MovieDataChunkIterator(movie_file), compression="gzip")
+        img_srs = ImageSeries(name="imageseries", data=it, unit="na", starting_time=None, rate=1.0)
+        self.nwbfile.add_acquisition(img_srs)
+        with NWBHDF5IO(path=self.nwbfile_path, mode="w") as io:
+            io.write(self.nwbfile)
+            print(self.nwbfile_path)
+        with NWBHDF5IO(path=self.nwbfile_path, mode="r") as io:
+            nwbfile = io.read()
+            assert nwbfile.acquisition["imageseries"].data.chunks == (1,) + self.frame_shape
```

Comment:
I would highly recommend adding a full array assertion to make sure the frames out are equal to the frames in.

Reply:
for tuples containing ints, I dont think `assert a==b` and `assert all([a[i]==j for i,j in enumerate(b)])` are any different

Reply:
Sorry for any confusion. I mean full array assertions on the `data` itself not just the chunks. Similar to how we examine full traces in the GIN `ecephys` testing data.

Reply:
The original way of testing the `.data.chunks` field is fine